crypto/crypto.go

```diff
@@ -151,6 +152,19 @@ func GenerateKey() (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(S256(), rand.Reader)
 }
 
+func ValidateSignatureValues(v byte, r, s *big.Int) bool {
+	secp256k1n := common.String2Big("0xfffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141")
```

Comment:
Move this outside of the function so it doesn't need to be converted and initialised for each function call.